src/ttsim/interface_dag_elements/results.py

```diff
@@ -14,29 +15,36 @@
 if TYPE_CHECKING:
     from ttsim.typing import (
         FlatData,
+        IntColumn,
         NestedData,
         NestedStrings,
         QNameData,
     )
 
 
 @interface_function()
-def tree(raw_results__combined: QNameData, input_data__flat: FlatData) -> NestedData:
-    """The combined results as a tree.
+def tree(
+    raw_results__columns: QNameData,
+    raw_results__params: QNameData,
+    raw_results__from_input_data: QNameData,
+    input_data__sort_indices: IntColumn,
+) -> NestedData:
+    """The combined results as a tree with original row order restored."""
+    restore_order = numpy.empty(len(input_data__sort_indices), dtype=int)
+    restore_order[input_data__sort_indices] = numpy.arange(
+        len(input_data__sort_indices)
+    )
 
-    The transformed id's are converted back to their original values.
+    def reorder_arrays(v: object) -> object:
```

Comment:
```suggestion
    def reorder_arrays(v: Any) -> Any:
```
Just because we are using this one in other places of the code.

Reply:
Ah, got it, just needed to add `  # noqa: ANN401`

Reply:
Just append `# noqa: ANN401` to the end of the line. Should be plenty of examples in the codebase.